### PR TITLE
Remove deprecated ScrollBehavior.buildViewportChrome

### PR DESCRIPTION
**Pull request by Piinks**
This removes `ScrollBehavior.buildViewportChrome`

This deprecated class has reached end of life after the release of Flutter 3.3

- The replacement is to use `ScrollBehavior.buildOverscrollIndicator` instead.
-  ✅  Migration via dart fix supported
- 🕐 This API was deprecated in
  - https://github.com/flutter/flutter/pull/78588, tagged v2.1

Part of https://github.com/flutter/flutter/issues/111708
For the full list of deprecations being removed in this batch, see [flutter.dev/go/deprecations-removed-after-3-3](https://github.com/flutter/flutter/pull/flutter.dev/go/deprecations-removed-after-3-3)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by Piinks** [comment]
Running internal testing to validate this is ready.

**Comment by Piinks** [comment]
Customers are migrated and ready

File: packages/flutter/lib/src/widgets/scroll_configuration.dart [hunk 1]
```diff
@@ -193,11 +154,30 @@ class ScrollBehavior {
   /// Applies a [GlowingOverscrollIndicator] to the child widget on
   /// [TargetPlatform.android] and [TargetPlatform.fuchsia].
   Widget buildOverscrollIndicator(BuildContext context, Widget child, ScrollableDetails details) {
-    // TODO(Piinks): Move implementation from buildViewportChrome here after
-    //  deprecation period
-    // When modifying this function, consider modifying the implementation in
-    // the Material and Cupertino subclasses as well.
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/scroll_configuration.dart:199]: Should we keep this comment? That's still true and relevant, right?

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/widgets/scroll_configuration.dart (hunk 1 above)
Comment (Piinks) [reply packages/flutter/lib/src/widgets/scroll_configuration.dart:199]: Oh yeah definitely. Let me update that across subclasses as well while we are here. Thanks!

**Review by ghost — COMMENTED** [review]
packages/flutter/lib/fix_data.yaml

**Pull request merged**